Sources/Wendy/cli/commands/OSCommand.swift

```diff
@@ -428,15 +428,14 @@ struct OSCommand: AsyncParsableCommand {
 
             // Download and extract as separate progress bars when not using cache
             let imageDownloader = ImageDownloaderFactory.createImageDownloader()
-            let realDownloader = imageDownloader as! ImageDownloader
 
             var localImagePath: String
 
             // Check if cached image exists and matches the latest version
-            let cachedImagePath = await realDownloader.cachedImageIfValid(
+            let cachedImagePath = try await imageDownloader.cachedImageIfValid(
```

Comment:
is a failure here supposed to fail the command all together or just redownload? what are you thinking?